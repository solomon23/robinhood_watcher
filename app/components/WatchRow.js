// @flow
import React, { Component } from 'react'
import classnames from 'classnames'
import { ipcRenderer } from 'electron'
import { USD } from '../services/utils'
import styles from './styles/StockRow.scss'

type Props = {
  stock: object
}

export default class WatchRow extends Component<Props> {
  props: Props

  static onClick(symbol) {
    ipcRenderer.send('CHART', { symbol })
  }

  render() {
    const { stock } = this.props

    const price = stock.last_extended_hours_trade_price || stock.last_trade_price
    const oldPrice = stock.previous_close

    const dif = (price - oldPrice)
    const difference = `${dif > 0 ? '+' : ''}${USD(dif)}`

    return (
      <div className={styles.stockRow} onClick={() => WatchRow.onClick(stock.symbol)}>
        <div className={styles.sybmol}>
          {stock.symbol}
        </div>
        <div className={classnames(styles.value, { [styles.loss]: dif < 0 })}>
          {difference}
        </div>
      </div>
    )
  }
}
