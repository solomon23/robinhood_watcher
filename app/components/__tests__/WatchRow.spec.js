import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import WatchRow from '../WatchRow'

Enzyme.configure({ adapter: new Adapter() })

const actions = {}
const defaultState = {
  stock: {},
  viewBy: 'TOTAL',
}

describe('WatchRow component', () => {
  beforeEach(() => {
    defaultState.stock = { 'symbol': 'a', last_trade_price: 10, previous_close: 5 }
  })

  it('should display positive value', () => {
    const component = shallow(<WatchRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should display negative value', () => {
    defaultState.stock.previous_close = 20
    const component = shallow(<WatchRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should display after hours value', () => {
    defaultState.stock.last_extended_hours_trade_price = 20
    const component = shallow(<WatchRow {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
