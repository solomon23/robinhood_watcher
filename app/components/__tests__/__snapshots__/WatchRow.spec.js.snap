// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchRow component should display after hours value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
  </div>
  <div
    className="value"
  >
    +$15.00
  </div>
</div>
`;

exports[`WatchRow component should display negative value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
  </div>
  <div
    className="value loss"
  >
    -$10.00
  </div>
</div>
`;

exports[`WatchRow component should display positive value 1`] = `
<div
  className="stockRow"
  onClick={[Function]}
>
  <div
    className="sybmol"
  >
    a
  </div>
  <div
    className="value"
  >
    +$5.00
  </div>
</div>
`;
